client/src/components/uis/StatusMessageView.tsx

```diff
@@ -0,0 +1,136 @@
+import {Animated, TouchableOpacity} from 'react-native';
+import React, {FC, MutableRefObject, useEffect, useRef, useState} from 'react';
+
+import {Icon} from 'dooboo-ui';
+import styled from '@emotion/native';
+
+type StyledTextProps = {
+  opened: Boolean;
+};
+
+type Props = {
+  statusMessage: String;
+  transitionOpacity: MutableRefObject<Animated.Value>;
+  modalLayout: {
+    width: number;
+    height: number;
+  };
+  opened: boolean;
+  handleAnim: () => void;
+};
+
+const StyledTextstatusMessage = styled.Text<StyledTextProps>`
+  font-size: 12px;
+  color: white;
+  align-self: center;
+  font-weight: ${(props) => (props.opened ? 'bold' : 'normal')};
+`;
+
+const MAX_STATUS_MESSAGE_LINES = 10;
+
+const StatusMessageView: FC<Props> = (props: Props) => {
+  const [bodyHeight, setBodyHeight] = useState(0);
+
+  const [seeMore, setSeeMore] = useState<{show: Boolean; length: number}>({
+    show: false,
+    length: 0,
+  });
+
+  const [textLayoutWidth, setTextLayoutWidth] = useState(0);
+
+  const transition = useRef(new Animated.Value(140)).current;
```

Comment:
Indeed, it is the first time I wrote animation codes for myself, lots of online references suggest to use `useRef` regardless of `using ref` because of (I think) re-rendering issue. 
https://blog.codecentric.de/en/2019/07/react-native-animated-with-hooks/#:~:text=a%20different%20solution.-,useRef,-useRef%20returns%20a

https://reactjs.org/docs/hooks-reference.html#useref
also mention use case of `useRef` regardless of `using ref`.